### PR TITLE
Fixes #25767 - fixed call to xen() after rebase

### PR DESCRIPTION
**Pull request by ShimShtein**
Another leftover of https://github.com/theforeman/foreman/pull/5683.

### CODE REVIEW

File: lib/foreman/renderer/scope/variables/base.rb [hunk 1]
```diff
@@ -92,7 +92,7 @@ def yast_attributes
 
           def xenserver_attributes
             @mediapath = mediumpath(@medium_provider) if @medium_provider
-            @xen = xen(arch)
+            @xen = xen(@medium_provider)
```
Comment (mmoll) [inline lib/foreman/renderer/scope/variables/base.rb:96]: is there also `if @medium_provider` needed here?

**Comment by mmoll** [comment]
Can this somehow get tested? I thought the template snaphots would take care of such cases?

**Comment by tbrisker** [comment]
@mmoll snapshot tests are currently only run with a RHEL host so we miss all of these - https://projects.theforeman.org/issues/25613 is open to fix that, but we would still only be able to test OSes we ship templates for (so the issues with windows wouldn't be found for example)

**Comment by ShimShtein** [comment]
@mmoll, now the code is more defensive. It won't throw an exception if medium_provider does not exist.

**Review by tbrisker — APPROVED** [review]
Thanks @ShimShtein !

**Comment by tbrisker** [comment]
1.20 - 319eabe71

**Pull request merged**